OpenRA.Mods.Common/Traits/DetectCloaked.cs

```diff
@@ -24,9 +26,29 @@ public class DetectCloakedInfo : ConditionalTraitInfo
 		public override object Create(ActorInitializer init) { return new DetectCloaked(this); }
 	}
 
-	public class DetectCloaked : ConditionalTrait<DetectCloakedInfo>
+	public class DetectCloaked : ConditionalTrait<DetectCloakedInfo>, INotifyCreated
 	{
+		IDetectCloakedModifier[] rangeModifiers;
+
 		public DetectCloaked(DetectCloakedInfo info)
 			: base(info) { }
+
+		void INotifyCreated.Created(Actor self)
+		{
+			rangeModifiers = self.TraitsImplementing<IDetectCloakedModifier>().ToArray();
+		}
+
+		public WDist Range
+		{
+			get
+			{
+				if (IsTraitDisabled)
+					return WDist.Zero;
+
+				var revealsShroudModifier = rangeModifiers.Select(x => x.GetDetectCloakedModifier());
```

Comment:
`revealsShroudModifier` is probably copy-paste, please change that to something like `detectCloakedModifier`.